frontend/src/scenes/feature-flags/activityDescriptions.tsx

```diff
@@ -142,25 +119,60 @@ const featureFlagActionsMapping: {
     is_simple_flag: () => null,
 }
 
-export function flagActivityDescriber(logItem: ActivityLogItem): (string | JSX.Element | null)[] {
+export function flagActivityDescriber(logItem: ActivityLogItem): string | JSX.Element | null {
     if (logItem.scope != 'FeatureFlag') {
-        return [] // currently, only humanizes the feature flag scope
+        console.error('feature flag decsriber received a non-feature flag activity')
+        return null // only humanizes the feature flag scope
     }
-    const descriptions = []
+
     if (logItem.activity == 'created') {
-        descriptions.push(<>created the flag: {nameOrLinkToFlag(logItem)}</>)
+        return <>created the flag: {nameOrLinkToFlag(logItem)}</>
     }
     if (logItem.activity == 'deleted') {
-        descriptions.push(<>deleted the flag: {logItem.detail.name}</>)
+        return <>deleted the flag: {logItem.detail.name}</>
     }
     if (logItem.activity == 'updated') {
+        let changes: (string | JSX.Element | null)[] = []
+
         for (const change of logItem.detail.changes || []) {
             if (!change?.field) {
-                continue // model changes have to have a "field" to be described
+                continue // feature flag updates have to have a "field" to be described
             }
 
-            descriptions.push(featureFlagActionsMapping[change.field](logItem, change))
+            changes = changes.concat(featureFlagActionsMapping[change.field](change))
+        }
+
+        if (changes.length) {
+            return <SentenceList listParts={changes} suffix={<>on {nameOrLinkToFlag(logItem)}</>} />
         }
     }
-    return descriptions
+
+    return null
+}
+
+interface SentenceListProps {
+    listParts: (string | JSX.Element | null)[]
+    prefix?: string | JSX.Element | null
+    suffix?: string | JSX.Element | null
+}
+
+function SentenceList({ listParts, prefix = null, suffix = null }: SentenceListProps): JSX.Element {
+    return (
+        <div className="sentence-list">
+            {prefix && <div>{prefix}&nbsp;</div>}
+            <>
+                {listParts.flatMap((part, index, all) => {
+                    const isntFirst = index > 0
+                    const isLast = index === all.length - 1
+                    const atLeastThree = all.length >= 2
+                    return [
+                        isntFirst && <div key={`${index}-a`}>,&nbsp;</div>,
+                        isLast && atLeastThree && <div key={`${index}-b`}>and&nbsp;</div>,
+                        <div key={`${index}-c`}>{part}</div>,
+                    ]
+                })}
+            </>
+            {suffix && <div>&nbsp;{suffix}</div>}
+        </div>
+    )
 }
```

Comment:
This `SentenceList` component is quite interesting and I've probably implemented equivalents of it a few times already (in `InsightCard` for instance). Non-blocking, but it could be super useful to extract it out into `lib/components` for general use.

Reply:
Cool, I'll look at yours too.

It's still not right. E.g. cos everything is `inline` the highlighted blocks don't have full height :/